go/libraries/doltcore/sqle/enginetest/dolt_queries_schema_merge.go

```diff
@@ -2003,3 +2094,236 @@ var SchemaChangeTestsSchemaConflicts = []MergeScriptTest{
+			},
+		},
+	},
+	{
+		Name: "adding a stored generated column to one side",
+		AncSetUpScript: []string{
+			"create table t (pk int primary key);",
+			"insert into t (pk) values (1);",
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 1) stored",
+			"insert into t (pk) values (3);",
+			"alter table t add index idx1 (col1, pk);",
+			"alter table t add index idx2 (col1);",
+		},
+		LeftSetUpScript: []string{
+			"insert into t (pk) values (2);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+			},
+			{
+				Query: "select pk, col1 from t;",
+				Expected: []sql.Row{
+					{1, 2},
+					{2, 3},
+					{3, 4},
+				},
+			},
+		},
+	},
+	{
+		Name: "adding generated columns to both sides",
+		AncSetUpScript: []string{
+			"create table t (pk int primary key);",
+			"insert into t values (1), (2);",
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col2 varchar(100) as (concat(pk, 'hello'));",
+			"insert into t (pk) values (3), (4);",
+			"alter table t add index (col2);",
+		},
+		LeftSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 100) stored;",
+			"insert into t (pk) values (5), (6);",
+			"alter table t add index (col1);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+				Skip:     true, // this fails merging right into left
+			},
+			{
+				Query: "select pk, col1, col2 from t;",
+				Expected: []sql.Row{
+					{1, 101, "1hello"},
+					{2, 102, "2hello"},
+					{3, 103, "3hello"},
+					{4, 104, "4hello"},
+					{5, 105, "5hello"},
+					{6, 106, "6hello"},
+				},
+				Skip: true, // this fails merging right into left
+			},
+		},
+	},
+	{
+		Name: "convergent schema changes with virtual columns",
+		AncSetUpScript: []string{
+			"set autocommit = 0;",
+			"CREATE table t (pk int primary key, col1 int);",
+			"INSERT into t values (1, 10);",
+		},
+		RightSetUpScript: []string{
+			"alter table t modify column col1 int not null;",
+			"alter table t add column col3 int as (pk + 1);",
+			"alter table t add index idx1 (col3, col1);",
+		},
+		LeftSetUpScript: []string{
+			"alter table t modify column col1 int not null;",
+			"alter table t add column col3 int as (pk + 1);",
+			"alter table t add index idx1 (col3, col1);",
+			"update t set col1=-1000 where t.pk = 1;",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+			},
+			{
+				Query: "show create table t;",
+				Expected: []sql.Row{{"t",
+					"CREATE TABLE `t` (\n" +
+						"  `pk` int NOT NULL,\n" +
+						"  `col1` int NOT NULL,\n" +
+						"  `col3` int GENERATED ALWAYS AS ((pk + 1)),\n" +
+						"  PRIMARY KEY (`pk`)\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
```

Comment:
Shouldn't the index `idx1` show up in here?

Reply:
Good catch